src/classes/LineItemTriggerHandler.cls

```diff
@@ -91,6 +93,7 @@ public with sharing class LineItemTriggerHandler {
 		public CategoryCounter(List<Line_Item__c> lineItems) {
 			for (Line_Item__c li : lineItems) { 
 				add(li.Order__c, li.Lot__r.Product_Category__c, li.Units__c);
+				update li;
```

Comment:
This DML statement is inside a constructor. Remove any data-altering statement from the constructor.